platform/darwin/test/MGLCodingTests.m

```diff
@@ -84,8 +100,24 @@ - (void)testPolygon {
     [NSKeyedArchiver archiveRootObject:polygon toFile:filePath];
 
     MGLPolygon *unarchivedPolygon = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
+    [unarchivedPolygon coordinate];
 
     XCTAssertEqualObjects(polygon, unarchivedPolygon);
+    
+    CLLocationCoordinate2D squareCoordinates[] = {
+        CLLocationCoordinate2DMake(100.0, 0.0),
+        CLLocationCoordinate2DMake(101.0, 0.0),
+        CLLocationCoordinate2DMake(101.0, 1.0),
+        CLLocationCoordinate2DMake(100.0, 1.0),
+    };
+    
+    NSUInteger squareCoordinatesCount = sizeof(squareCoordinates) / sizeof(CLLocationCoordinate2D);
+    MGLPolygon *squarePolygon = [MGLPolygon polygonWithCoordinates:squareCoordinates count:squareCoordinatesCount];
+    CLLocationCoordinate2D squareCenter = CLLocationCoordinate2DMake(100.5, 0.5);
```

Comment:
Do you think this is an adequate test of the added functionality? Do we need more unit tests specifically about the `coordinate` properties that this PR adds? Perhaps we could port some tests from Turf or from [GeometryTests.swift](https://github.com/mapbox/mapbox-navigation-ios/blob/5be6ada250e43804d3d74c5724687a761abb6271/MapboxCoreNavigationTests/GeometryTests.swift).